taskcluster/ci/ui-test/kind.yml

```diff
@@ -0,0 +1,50 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+---
+loader: taskgraph.loader.transform:loader
+
+transforms:
+    - fenix_taskgraph.transforms.test:transforms
+    - taskgraph.transforms.job:transforms
+    - taskgraph.transforms.task:transforms
+
+jobs:
+    x86:
```

Comment:
Thanks for the context! I took a closer look and I think you'd want to have a dedicated `x86` job. Here's my reasoning: 

1. In the current state, the job is called `ui-test-x86`. That's a standard scheme made by taskgraph: `{kind}-{job_name}`. If changed line 13, we would end up with the name `ui-test-ui-test`.
2. If you add a second test execution at line 27, you would end up with a sequential run: first, all the x86 test run and finish, then you kick off the ARM ones. What we can do is: we define 2 jobs: `x86` and `arm`, so both run at the very same time, and you don't have to wait for x86 to complete. 
3. We can factorize the common configuration, so creating a new ARM job doesn't require us to duplicate the 35 lines, but just to define what is different. 

I'm intrigued about `ui-test-device-compare`. What would this job do? I wonder if we can reuse the results of the `x86` and `arm` jobs discussed above. 

In any case, I understand you'd prefer to get this PR landed first. So I'm happy to continue the conversation in a follow up and keep the current code as is 🙂

Reply:
Yes, let's land this so we can make progress iteratively.  

`ui-test-device-compare` would be used for testing a very common issue with UI tests which is "flakiness."  Flakiness comes about when you have a test that runs perfectly 100% of the time on one or two diferent devices or APIs, but then fails a percentage of the time on others, or fails intermittently (often due to test timing issues).  We originally stopped using x86 devices because they seemed more prone to this kind of behavior, but google can't support the volume of real devices we need so we switched back.  In the meantime, we've identified a handful of tests that are important for coverage but are flaky (at least on x86 or so we think).  By segregating these and comparing x86 to arm, we could eventually decide to run a small number of tests on arm only and the rest on x86.  This is important also because, there are times when a test fails intermittently because fenix itself may have a race condition we want to become aware of so, keeping these tests is vital.